### DISCUSSION TITLE
Plugins Sublime

### DISCUSSION THREAD

**Discussion by willianjusten** [frontendbr/forum#93] [Perguntas e Respostas]
Puxando [meu post](http://willianjusten.com.br/meus-plugins-favoritos-do-sublime-text/) sobre plugins no Sublime.

E também um tweet do @LFeh com os [plugins dele](https://github.com/LFeh/sublime.config), fiquei curioso para saber o que a galera usa, posta aí =D

**Comment by viniciusdocanto** [comment]
Os meus são:
- [AlignTab](https://packagecontrol.io/packages/AlignTab)
- [Color Highlighter](https://packagecontrol.io/packages/Color%20Highlighter)
- [CSS3](https://packagecontrol.io/packages/CSS3)
- [Emmet](https://packagecontrol.io/packages/Emmet)
- [Git](https://packagecontrol.io/packages/Git)
- [Less](https://packagecontrol.io/packages/LESS)
- [Nodejs](https://packagecontrol.io/packages/Nodejs)
- [Grunt](https://packagecontrol.io/packages/Grunt)
- [Terminal](https://packagecontrol.io/packages/Terminal)

**Config:**

```
{
    "Seti_tabs_small": true,
    "color_scheme": "Packages/User/Color Highlighter/themes/Seti_monokai.tmTheme",
    "draw_minimap_border": true,
    "font_size": 8,
    "highlight_line": true,
    "highlight_modified_tabs": true,
    "margin": 0,
    "soda_classic_tabs": "true",
    "soda_folder_icons": "true",
    "theme": "Seti_orig.sublime-theme",
    "word_wrap": true
}
```

**Comment by rickbenetti** [comment]
Segue minha lista de plugins:
- [EditorConfig](https://github.com/sindresorhus/editorconfig-sublime)
- [Emmet](https://packagecontrol.io/packages/Emmet)
- [HTML-CSS-JS Prettify](https://github.com/victorporof/Sublime-HTMLPrettify)
- [Sidebar Enhancement](https://github.com/titoBouzout/SideBarEnhancements)
- [HTML5](https://packagecontrol.io/packages/HTML5)
- [HTML Attributes](https://packagecontrol.io/packages/HTMLAttributes)
- [HTML Beautify](https://packagecontrol.io/packages/HTMLBeautify)
- [LESS](https://packagecontrol.io/packages/LESS)
- [Material Theme](https://packagecontrol.io/packages/Material%20Theme)
- [nginx](https://packagecontrol.io/packages/nginx)
- [SassBeautify](https://packagecontrol.io/packages/SassBeautify)
- [Stylus](https://github.com/billymoon/Stylus)

Segue minha Config

```
{
    "always_prompt_for_file_reload": false,
    "always_show_minimap_viewport": true,
    "animation_enabled": true,
    "atomic_save": true,
    "auto_close_tags": true,
    "auto_complete": true,
    "auto_complete_commit_on_tab": false,
    "auto_complete_delay": 50,
    "auto_complete_selector": "source - comment, meta.tag - punctuation.definition.tag.begin",
    "auto_complete_size_limit": 4194304,
    "auto_complete_triggers":
    [
        {
            "characters": "<",
            "selector": "text.html"
        }
    ],
    "auto_complete_with_fields": false,
    "auto_find_in_selection": false,
    "auto_indent": true,
    "auto_match_enabled": true,
    "binary_file_patterns":
    [
        "*.jpg",
        "*.jpeg",
        "*.png",
        "*.gif",
        "*.ttf",
        "*.tga",
        "*.dds",
        "*.ico",
        "*.eot",
        "*.pdf",
        "*.swf",
        "*.jar",
        "*.zip"
    ],
    "bold_folder_labels": true,
    "caret_extra_bottom": 0,
    "caret_extra_top": 0,
    "caret_extra_width": 1,
    "caret_style": "phase",
    "close_windows_when_empty": false,
    "color_inactive_tabs": false,
    "color_scheme": "Packages/Material Theme/schemes/Material-Theme.tmTheme",
    "copy_with_empty_selection": false,
    "create_window_at_startup": true,
    "default_encoding": "UTF-8",
    "default_line_ending": "system",
    "detect_indentation": true,
    "drag_text": false,
    "draw_centered": false,
    "draw_indent_guides": true,
    "draw_minimap_border": false,
    "draw_white_space": "selection",
    "enable_hexadecimal_encoding": true,
    "enable_tab_scrolling": false,
    "enable_telemetry": "auto",
    "ensure_newline_at_eof_on_save": false,
    "fade_fold_buttons": true,
    "fallback_encoding": "UTF-8",
    "file_exclude_patterns":
    [
        "*.pyc",
        "*.pyo",
        "*.exe",
        "*.dll",
        "*.obj",
        "*.o",
        "*.a",
        "*.lib",
        "*.so",
        "*.dylib",
        "*.ncb",
        "*.sdf",
        "*.suo",
        "*.pdb",
        "*.idb",
        ".DS_Store",
        "*.class",
        "*.psd",
        "*.db",
        "*.sublime-workspace"
    ],
    "find_selected_text": true,
    "fold_buttons": true,
    "folder_exclude_patterns":
    [
        ".svn",
        ".git",
        ".hg",
        "CVS"
    ],
    "font_face": "Anonymous Pro",
    "font_options":
    [
        "gray_antialias",
        "subpixel_antialias"
    ],
    "font_size": 16,
    "gpu_window_buffer": "auto",
    "gutter": true,
    "highlight_line": true,
    "highlight_modified_tabs": true,
    "hot_exit": true,
    "ignored_packages":
    [
        "Vintage"
    ],
    "indent_guide_options":
    [
        "draw_normal",
        "draw_active"
    ],
    "indent_subsequent_lines": true,
    "indent_to_bracket": false,
    "index_exclude_patterns":
    [
        "*.log"
    ],
    "index_files": true,
    "line_numbers": true,
    "line_padding_bottom": 5,
    "line_padding_top": 5,
    "margin": 0,
    "match_selection": true,
    "match_tags": true,
    "material_theme_appbar_yellow": true,
    "material_theme_bold_tab": true,
    "material_theme_compact_sidebar": false,
    "material_theme_contrast_mode": false,
    "material_theme_disable_fileicons": false,
    "material_theme_disable_folder_animation": false,
    "material_theme_disable_tree_indicator": false,
    "material_theme_panel_separator": true,
    "material_theme_small_statusbar": true,
    "material_theme_small_tab": true,
    "material_theme_tabs_autowidth": true,
    "material_theme_tabs_separator": true,
    "material_theme_tree_headings": true,
    "move_to_limit_on_up_down": false,
    "open_files_in_new_window": false,
    "overlay_scroll_bars": "enabled",
    "preview_on_click": false,
    "remember_open_files": true,
    "rulers":
    [
        80,
        120,
        160
    ],
    "save_on_focus_lost": false,
    "scroll_past_end": true,
    "scroll_speed": 5.0,
    "shift_tab_unindent": false,
    "show_encoding": false,
    "show_full_path": false,
    "show_line_endings": false,
    "show_panel_on_build": true,
    "show_tab_close_buttons": true,
    "sidebar_default": "large",
    "smart_indent": true,
    "spell_check": false,
    "tab_completion": true,
    "tab_size": 4,
    "theme": "Material-Theme.sublime-theme",
    "translate_tabs_to_spaces": true,
    "tree_animation_enabled": true,
    "trim_automatic_white_space": true,
    "trim_trailing_white_space_on_save": true,
    "use_simple_full_screen": false,
    "use_tab_stops": true,
    "word_separators": "./\\()\"'-:,.;<>~!@#$%^&*|+=[]{}`~?\\.",
    "word_wrap": true,
    "wrap_width": 0
}
```

**Comment by ghost** [comment]
A minha está [aqui](https://github.com/meche/sublime-text)

**Comment by fdaciuk** [comment]
Não uso mais Sublime, mas quando usava, era essa listinha: https://gist.github.com/fdaciuk/8b3269a3d797ee723672

:grin:

**Comment by wellingtonpgp** [comment]
Eu não estou usando sublime mais, estou utilizando o NetBeans depois que ele começou a dar suporte para Node.js ficou tudo lindo já utilizava ele em projetos java a um bom tempo, gravei todos atalhos, mas tenho sublime ali para uma coisa mais rápida, como todos sabem IDE costuma ser mais lenta para iniciar.

**Comment by jhonnymoreira** [comment]
## Plugins
- AlignTab
- Color Highlighter
- Dictionaries
- EditorConfig
- Emmet
- Git Conflict Resolver
- GitGutter
- JavaScript Next - ES6 Syntax
- Material Theme
- Origami
- PackageResourceViewer
- RubyNext
- Sass
- SCSS
- SideBarEnhancements
- Stylus

## Config

```
{
  "always_show_minimap_viewport": true,
  "bold_folder_labels": true,
  "color_scheme": "Packages/Material Theme/schemes/Material-Theme.tmTheme",
  "dictionary": "Packages/Language - English/en_US.dic",
  "font_options":
  [
    "gray_antialias"
  ],
  "font_size": 12,
  "ignored_packages":
  [
    "JavaScript",
    "Ruby",
    "Vintage"
  ],
  "indent_guide_options":
  [
    "draw_active"
  ],
  "line_padding_bottom": 3,
  "line_padding_top": 3,
  "material_theme_accent_indigo": true,
  "material_theme_contrast_mode": true,
  "overlay_scroll_bars": "enabled",
  "rulers":
  [
    50,
    72,
    80
  ],
  "tab_size": 2,
  "theme": "Material-Theme.sublime-theme",
  "translate_tabs_to_spaces": true
}
```

**Comment by brunoravanhani** [comment]
Para linter de javascript o melhor é o Sublime Linter com algum outro plugin de linter como o jscs ou jshint? Ou alguém usa algum outro?

**Comment by twobanks** [comment]
# Plugins
- Alignment
- AutoFileName
- Color Highlighter
- CSS3
- DocBlockr
- Emmet
- Git
- EsFormatter
- Gulp
- SublimeLinter

# Temas
- Material Theme
- Theme - itg.flat

# Configurações

{
    "always_show_minimap_viewport": true,
    "bold_folder_labels": true,
    "color_scheme": "Packages/Material Theme/schemes/Material-Theme.tmTheme",
    "font_face": "Menlo",
    "font_options":
    [
        "gray_antialias",
        "subpixel_antialias"
    ],
    "font_size": 12,
    "ignored_packages":
    [
        "Vintage"
    ],
    "indent_guide_options":
    [
        "draw_normal",
        "draw_active"
    ],
    "line_padding_bottom": 3,
    "line_padding_top": 3,
    "material_theme_bold_tab": true,
    "material_theme_compact_sidebar": true,
    "material_theme_contrast_mode": true,
    "material_theme_disable_fileicons": true,
    "material_theme_disable_folder_animation": true,
    "material_theme_disable_tree_indicator": true,
    "material_theme_panel_separator": true,
    "material_theme_small_statusbar": true,
    "material_theme_small_tab": true,
    "material_theme_tabs_autowidth": true,
    "material_theme_tabs_separator": true,
    "material_theme_tree_headings": true,
    "overlay_scroll_bars": "enabled",
    "theme": "Material-Theme.sublime-theme"
}

**Comment by rubenmarcus** [comment]
alguem sabe como configurar o phpfm no windows? Instalei o PHP no windows, configurei conforme fala no site, mas ele nao formata de jeito nenhum, to usando o PHP Storm pra isso,mas e meio pesado aqui p minha maquina
